Sources/TensorFlow/Operators/Math.swift

```diff
@@ -1399,9 +1399,9 @@ internal func _vjpMinMaxHelper<T: TensorFlowFloatingPoint>(
     originalValue: Tensor<T>,
     seed: Tensor<T>
 ) -> (Tensor<T>, Tensor<T>) {
-    let denominator = 1 + Tensor<T>(x .== y)
-    let lhsGrad = seed * Tensor<T>(x .== originalValue) / denominator
-    let rhsGrad = seed * Tensor<T>(y .== originalValue) / denominator
+    let tieBreakerMask = Tensor<T>(x .!= y)
```

Comment:
Are you sure this implementation is correct?

Here's my result, after porting directly from [`MaximumMinimumGradHelper` in `tensorflow/core/ops/math_grad.cc`](https://github.com/tensorflow/tensorflow/blob/dac733509fbe366804656a71f15ba0712bb5e69d/tensorflow/core/ops/math_grad.cc#L614):
```diff
diff --git a/Sources/TensorFlow/Operators/Math.swift b/Sources/TensorFlow/Operators/Math.swift
index 81d890f..c8688af 100644
--- a/Sources/TensorFlow/Operators/Math.swift
+++ b/Sources/TensorFlow/Operators/Math.swift
@@ -1344,7 +1344,7 @@ internal func _vjpMax<T: TensorFlowFloatingPoint>(
     _ y: Tensor<T>
 ) -> (Tensor<T>, (Tensor<T>) -> (Tensor<T>, Tensor<T>)) {
     let value = max(x, y)
-    return (value, { v in _vjpMinMaxHelper(x, y, originalValue: value, seed: v) })
+    return (value, { v in _vjpMinMaxHelper(x, y, originalValue: value, seed: v, comparisonOperation: .<=) })
 }

 /// Returns the element-wise maximum of the scalar and the tensor, broadcasting the scalar.
@@ -1375,7 +1375,7 @@ internal func _vjpMin<T: TensorFlowFloatingPoint>(
     _ y: Tensor<T>
 ) -> (Tensor<T>, (Tensor<T>) -> (Tensor<T>, Tensor<T>)) {
     let value = min(x, y)
-    return (value, { v in _vjpMinMaxHelper(x, y, originalValue: value, seed: v) })
+    return (value, { v in _vjpMinMaxHelper(x, y, originalValue: value, seed: v, comparisonOperation: .>=) })
 }

 /// Returns the element-wise minimum of the scalar and the tensor, broadcasting the scalar.
@@ -1397,11 +1397,12 @@ internal func _vjpMinMaxHelper<T: TensorFlowFloatingPoint>(
     _ x: Tensor<T>,
     _ y: Tensor<T>,
     originalValue: Tensor<T>,
-    seed: Tensor<T>
+    seed: Tensor<T>,
+    comparisonOperation: (Tensor<T>, Tensor<T>) -> Tensor<Bool>
 ) -> (Tensor<T>, Tensor<T>) {
-    let denominator = 1 + Tensor<T>(x .== y)
-    let lhsGrad = seed * Tensor<T>(x .== originalValue) / denominator
-    let rhsGrad = seed * Tensor<T>(y .== originalValue) / denominator
+    let mask: Tensor<T> = Raw.cast(comparisonOperation(x, y))
+    let lhsGrad = seed * mask
+    let rhsGrad = seed - lhsGrad
     let (lhsShape, rhsShape) = (x.shapeTensor, y.shapeTensor)
     let (lhsAxes, rhsAxes) = Raw.broadcastGradientArgs(s0: lhsShape, s1: rhsShape)
     return (lhsGrad.sum(squeezingAxes: lhsAxes).reshaped(toShape: lhsShape),
```

Reply:
I couldn't get your solution working under the tests I've written. I'll look into it more.
My idea was just to multiply out the ties from the `rhsGrad` with the mask which doesn't seem efficient enough now that you point it out.

Reply:
@dan-zheng Ah I got it :smile:! I had misread a piece. Thank you!